var/spack/repos/builtin/packages/exawind/package.py

```diff
@@ -16,59 +16,88 @@ class Exawind(CMakePackage, CudaPackage, ROCmPackage):
-    variant("stk_simd", default=False, description="Enable SIMD in STK")
+    version("master", branch="main", submodules=True, preferred=True)
+    version("1.0.0", tag="v1.0.0", submodules=True)
+
+    variant("amr_wind_gpu", default=False, description="Enable AMR-Wind on the GPU")
+    variant("nalu_wind_gpu", default=False, description="Enable Nalu-Wind on the GPU")
     variant("umpire", default=False, description="Enable Umpire")
-    variant("tiny_profile", default=False, description="Turn on AMR-wind with tiny profile")
     variant("sycl", default=False, description="Enable SYCL backend for AMR-Wind")
     variant("gpu-aware-mpi", default=False, description="gpu-aware-mpi")
 
-    conflicts("amr-wind+hypre", when="+sycl")
-
     for arch in CudaPackage.cuda_arch_values:
-        depends_on("amr-wind+cuda cuda_arch=%s" % arch, when="+cuda cuda_arch=%s" % arch)
-        depends_on("nalu-wind+cuda cuda_arch=%s" % arch, when="+cuda cuda_arch=%s" % arch)
+        depends_on(
+            "amr-wind+cuda cuda_arch=%s" % arch, when="+amr_wind_gpu+cuda cuda_arch=%s" % arch
+        )
+        depends_on(
+            "nalu-wind+cuda cuda_arch=%s" % arch, when="+nalu_wind_gpu+cuda cuda_arch=%s" % arch
+        )
+        depends_on(
+            "trilinos+cuda cuda_arch=%s" % arch, when="+nalu_wind_gpu+cuda cuda_arch=%s" % arch
+        )
 
     for arch in ROCmPackage.amdgpu_targets:
-        depends_on("amr-wind+rocm amdgpu_target=%s" % arch, when="+rocm amdgpu_target=%s" % arch)
-        depends_on("nalu-wind+rocm amdgpu_target=%s" % arch, when="+rocm amdgpu_target=%s" % arch)
-
-    depends_on("nalu-wind+tioga")
-    depends_on("amr-wind+netcdf+mpi")
-    depends_on("tioga~nodegid")
+        depends_on(
+            "amr-wind+rocm amdgpu_target=%s" % arch,
+            when="+amr_wind_gpu+rocm amdgpu_target=%s" % arch,
+        )
+        depends_on(
+            "nalu-wind+rocm amdgpu_target=%s" % arch,
+            when="+nalu_wind_gpu+rocm amdgpu_target=%s" % arch,
+        )
+        depends_on(
+            "trilinos+rocm amdgpu_target=%s" % arch,
+            when="+nalu_wind_gpu+rocm amdgpu_target=%s" % arch,
+        )
+
+    depends_on("nalu-wind+hypre+fsi+openfast+tioga")
+    depends_on("amr-wind+netcdf+mpi+tiny_profile")
+    depends_on("trilinos")
     depends_on("yaml-cpp@0.6:")
-    depends_on("nalu-wind+openfast", when="+openfast")
-    depends_on("amr-wind+hypre", when="+hypre~sycl")
-    depends_on("amr-wind~hypre", when="~hypre")
-    depends_on("nalu-wind+hypre", when="+hypre")
-    depends_on("nalu-wind~hypre", when="~hypre")
-    depends_on("amr-wind+sycl", when="+sycl")
+    depends_on("tioga~nodegid")
+    depends_on("openfast+cxx@2.6.0:")
+    depends_on("amr-wind+sycl", when="+amr_wind_gpu+sycl")
+    depends_on("kokkos-nvcc-wrapper", type="build", when="+cuda")
+    depends_on("mpi")
     depends_on("nalu-wind+umpire", when="+umpire")
     depends_on("amr-wind+umpire", when="+umpire")
```

Comment:
This one is still kind of bugging me. I think it should be 
```
depends_on("amr-wind+umpire, when="+umpire ^amr-wind+hypre")
```

Otherwise doing mixed device will give symbol clashes if you use `+umpire` due to the multiple builds of `hypre` that would be required.

It makes things a little more pedantic when setting up `exawind` with `amr-wind+hypre`, but hopefully it won't be required too often.

Reply:
Agreed